Makefile

```diff
@@ -213,24 +217,46 @@ endif
 	gzip -9 < $^ > $@.tmp
 	mv $@.tmp $@
 
-nixpkgs/packages.json: packages-config.nix
-	nixpkgs=$$(nix-instantiate --find-file nixpkgs -I nixpkgs=$(NIXPKGS)); \
+nixpkgs/packages-nixos-$(NIXOS_SERIES).json: packages-config.nix
+	nixpkgs=$$(nix-instantiate --find-file nixpkgs -I nixpkgs=$(CHANNEL_NIXOS_STABLE)); \
 	(echo -n '{ "commit": "' && (cat $$nixpkgs/.git-revision || printf "unknown") && echo -n '","packages":' \
-	  && nix-env -f '<nixpkgs>' -I nixpkgs=$(NIXPKGS) -qa --json --arg config 'import ./packages-config.nix' \
+	  && nix-env -f '<nixpkgs>' -I nixpkgs=$(CHANNEL_NIXOS_STABLE) -qa --json --arg config 'import ./packages-config.nix' \
 	  && echo -n '}') \
 	  | sed "s|$$nixpkgs/||g" | jq -c . > $@.tmp
 	python -mjson.tool < $@.tmp > /dev/null
 	mv $@.tmp $@
 
-nixpkgs/packages-unstable.json: packages-config.nix
-	nixpkgs=$$(nix-instantiate --find-file nixpkgs -I nixpkgs=$(NIXPKGS_UNSTABLE)); \
+nixpkgs/packages-nixos-unstable.json: packages-config.nix
+	nixpkgs=$$(nix-instantiate --find-file nixpkgs -I nixpkgs=$(CHANNEL_NIXOS_UNSTABLE)); \
 	(echo -n '{ "commit": "' && (cat $$nixpkgs/.git-revision || printf "unknown") && echo -n '","packages":' \
-	  && nix-env -f '<nixpkgs>' -I nixpkgs=$(NIXPKGS_UNSTABLE) -qa --json --arg config 'import ./packages-config.nix' \
+	  && nix-env -f '<nixpkgs>' -I nixpkgs=$(CHANNEL_NIXOS_UNSTABLE) -qa --json --arg config 'import ./packages-config.nix' \
 	  && echo -n '}') \
 	  | sed "s|$$nixpkgs/||g" | jq -c . > $@.tmp
 	python -mjson.tool < $@.tmp > /dev/null
 	mv $@.tmp $@
 
+nixpkgs/packages-nixpkgs-unstable.json: packages-config.nix
+	nixpkgs=$$(nix-instantiate --find-file nixpkgs -I nixpkgs=$(CHANNEL_NIXPKGS_UNSTABLE)); \
+	(echo -n '{ "commit": "' && (cat $$nixpkgs/.git-revision || printf "unknown") && echo -n '","packages":' \
+	  && nix-env -f '<nixpkgs>' -I nixpkgs=$(CHANNEL_NIXPKGS_UNSTABLE) -qa --json --arg config 'import ./packages-config.nix' \
+	  && echo -n '}') \
+	  | sed "s|$$nixpkgs/||g" | jq -c . > $@.tmp
+	python -mjson.tool < $@.tmp > /dev/null
+	mv $@.tmp $@
+
+# Cute hack, this allows future expansion if desired
+# Mainly, this allows tracking NIXOS_SERIES
+nixpkgs/packages-channels.json: Makefile
+	echo '["nixos-$(NIXOS_SERIES)", "nixos-unstable", "nixpkgs-unstable"]' > $@
+
 nixos/options.json:
-	cat $$(nix-build --no-out-link '<nixpkgs/nixos/release.nix>' -I nixpkgs=$(NIXPKGS) -A options)/share/doc/nixos/options.json > $@.tmp
+	cat $$(nix-build --no-out-link '<nixpkgs/nixos/release.nix>' -I nixpkgs=$(CHANNEL_NIXOS_STABLE) -A options)/share/doc/nixos/options.json > $@.tmp
 	mv $@.tmp $@
+
+# This is a fine enough approximation of the dependencies
+# The `node_modules` folder will not be present at deployment, only for development.
+EXPLORER_JS = $(shell find packages-explorer/ -not -path 'packages-explorer/node_modules/*')
+
+nixos/packages-explorer.js: $(EXPLORER_JS)
+	(cd packages-explorer ; nix-build -I nixpkgs=$(CHANNEL_NIXOS_STABLE))
```

Comment:
Hm, I'd prefer not to call `nix-build` from the Makefile because that makes it harder to have a reproducible build of the homepage. 

For background, the `flake` branch of `nixos-homepage` adds a Nix build that calls the Makefile (see https://github.com/NixOS/nixos-homepage/blob/flake/flake.nix), so if the Makefile calls Nix, that won't work anymore.

Reply:
I do agree. I initially asked for help in figuring out a nix-build-free solution, but was told to continue with it. Though I think it should be possible to do the same thing that is done with the documentation updates I've been working on with @grahamc. Basically, have a full-fledged derivation that is symlinked in place during the nix build, and nix-built otherwise. It should be fixable relatively easily.

Reply:
This was pretty easy to fix, `flake.nix` now imports/builds the `package-explorer` Nix expression directly and passes the build result to the Makefile(https://github.com/NixOS/nixos-homepage/blob/762285da62e12280233feaf1f0c802b621e2313b/flake.nix#L44).